### PR TITLE
dataframe.redivide

### PR DESCRIPTION
**Pull request by mrocklin**
Analogous to `da.rechunk`, `dd.redivide` splits a dataframe across a new set of partitions

```python
In [1]: import pandas as pd

In [2]: df = pd.DataFrame({'x': [1, 2, 3, 4, 5, 6, 7, 8]})

In [3]: import dask.dataframe as dd

In [4]: a = dd.from_pandas(df, npartitions=4)

In [5]: b = a.redivide([0, 5, 6, 7])

In [6]: a.divisions
Out[6]: (0, 2, 4, 6, 7)

In [7]: b.divisions
Out[7]: (0, 5, 6, 7)

In [8]: b._visualize()
```

mydask

cc @cpcloud this may interest you

### CODE REVIEW

**Comment by mrocklin** [comment]
I recently renamed redivide to repartition.

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -963,4 +997,98 @@ def pd_split(df, p, seed=0):
...
+    Parameters
+    ----------
+    a: tuple
+        old divisions
+    b: tuple
+        new divisions
+    name: str
+        name of old dataframe
+    out: str
+        name of new dataframe
```
Comment (jcrist) [inline dask/dataframe/core.py]: `out` should be `out1` and `out2`.

**Comment by jcrist** [comment]
This looks good to me, but I'm fairly new to pandas.

File: dask/dataframe/core.py (hunk 1 above)
Comment (mrocklin) [inline dask/dataframe/core.py]: Fixed

**Pull request merged**